src/command/run.rs

```diff
@@ -92,6 +96,11 @@ impl Run {
             },
         };
 
+        let pnpm = match &self.pnpm {
+            None => InheritOption::None,
+            Some(version) => InheritOption::Some(pnpm::resolve(version.parse()?, session)?),
+        };
```

Comment:
Since `pnpm` isn't bundled with Node the way `npm` is, I believe this should mirror the `yarn` branch more than the `npm` one. Specifically, we should have an explicit `--no_pnpm` flag in the same way that disables the resolution inheritance, while omitting the value entirely leads to `InheritOption::Inherit`.